src/Altinn.Platform/Altinn.Platform.Storage/Storage/Helpers/AuthorizeInstancesHelper.cs

```diff
@@ -0,0 +1,305 @@
+
+        private static XacmlJsonMultiRequests CreateMultiRequestsCategory(List<XacmlJsonCategory> subjects, List<XacmlJsonCategory> actions, List<XacmlJsonCategory> resources)
+        {
+            List<string> subjectIds = subjects.Select(s => s.Id).ToList();
+            List<string> actionIds = actions.Select(a => a.Id).ToList();
+            List<string> resourceIds = resources.Select(r => r.Id).ToList();
+
+            XacmlJsonMultiRequests multiRequests = new XacmlJsonMultiRequests();
+            multiRequests.RequestReference = CreateRequestReference(subjectIds, actionIds, resourceIds);
+
+            return multiRequests;
+        }
+
+        private static List<XacmlJsonRequestReference> CreateRequestReference(List<string> subjectIds, List<string> actionIds, List<string> resourceIds)
+        {
+            List<XacmlJsonRequestReference> references = new List<XacmlJsonRequestReference>();
+
+            foreach (string resourceId in resourceIds)
+            {
+                foreach (string actionId in actionIds)
+                {
+                    foreach (string subjectId in subjectIds)
+                    {
+                        XacmlJsonRequestReference reference = new XacmlJsonRequestReference();
+                        List<string> referenceId = new List<string>();
+                        referenceId.Add(subjectId);
+                        referenceId.Add(actionId);
+                        referenceId.Add(resourceId);
+                        reference.ReferenceId = referenceId;
+                        references.Add(reference);
+                    }
+                }
+            }
+
+            return references;
+        }
+
+        /// <summary>
+        /// Will check if a instance is already in the list of authorized instances.
+        /// Returns the position to the inctance in the list if it is in the list, and -1 if not. 
+        /// </summary>
+        private int InstanceAlreadyAuthorized(List<MessageBoxInstance> authorizedInstances, Instance instance)
+        {
+            int instancePosition = 0; 
+            foreach (MessageBoxInstance authorizedInstance in authorizedInstances)
+            {
+                if (authorizedInstance.Id.Equals(instance.Id.Split("/")[1]))
+                {
+                    return instancePosition;
+                }
+
+                instancePosition++;
+            }
+
+            return -1;
+        }
+
+        /// <summary>
+        /// Authorize instances, and returns a list of MesseageBoxInstances with information about read and write rights of each instance. 
+        /// </summary>
+        public async Task<List<MessageBoxInstance>> AuthorizeMesseageBoxInstances(ClaimsPrincipal user, List<Instance> instances)
+        {
+            if (instances.Count <= 0)
+            {
+                return new List<MessageBoxInstance>();
+            }
+
+            List<MessageBoxInstance> authorizedInstances = new List<MessageBoxInstance>();
```

Comment:
Could we call this authorizedInstanceList ? Not the best name, but that would make it less simmilar to aurhorizedInstance